app/controllers/my-projects/project/hearing/add.js

```diff
@@ -0,0 +1,80 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import moment from 'moment';
+
+// create a hearing record when a user clicks on the submit hearing button
+function createHearing(thisHearing, hearingLocation, hearingDate) {
+  const newHearing = thisHearing.store.createRecord('hearing', {
+    project: thisHearing.model,
+    location: hearingLocation,
+    date: hearingDate,
+  });
+  newHearing.save();
+}
+
+export default class MyProjectsProjectHearingAddController extends Controller {
+  @service
+  store;
+
+  // for checking if user clicks submit button without inputting necessary attributes
+  dateMissing;
+
+  timeMissing;
+
+  locationMissing;
+
+  // options for ember-timepicker
+  timeOptions= {
+    dropdown: true,
+    interval: 5,
+    scrollbar: true,
+    minTime: '6:00 AM',
+    maxTime: '11:00 PM',
+    startTime: '6:00',
+  };
+
+  // minimum date for calendar
+  currentDate = new Date();
+
+  // for clearing search results on the geosearch input
+  onClearSearchResult = () => {};
+
+  // when a user clicks on the x in the input box
+  @action
+  clearSearchResult() {
+    this.onClearSearchResult();
+  }
+
+  // action triggered when click on "submit hearing" button
+  @action
+  saveHearingForm() {
+    // arguments to trigger a notification to the user that they need to input a field
+    if (!this.date) { this.set('dateMissing', true); } else { this.set('dateMissing', false); }
+    if (!this.location) { this.set('locationMissing', true); } else { this.set('locationMissing', false); }
+    if (!this.time) { this.set('timeMissing', true); } else { this.set('timeMissing', false); }
+
+    // dateUpdated = this.date + minutes & hours from this.time
+    const minutesAdded = moment(this.time).minute();
+    const hoursAdded = moment(this.time).hour();
+    const dateUpdated = moment(this.date).add(hoursAdded, 'h').add(minutesAdded, 'm').toDate();
+
+    // allows users to add additional location info, updates location string
+    let locationUpdated = '';
+
+    if (this.location && this.additionalLocationInfo) {
+      // locations come back as an object with geometry and bbl so
+      // we need to grab location.label to get address string
+      const locationLabel = this.location.label;
+      locationUpdated = locationLabel.concat('; ', this.additionalLocationInfo);
+    } else if (this.location) {
+      locationUpdated = this.location.label;
+    }
+
+    // if user has input date, location, and time then save to model and transition to to-review route
+    if (this.date && this.location && this.time) {
+      createHearing(this, locationUpdated, dateUpdated);
+      this.transitionToRoute('/my-projects/to-review');
+    }
+  }
+}
```

Comment:
This is great! One general comment I have about this: I would actually call `createRecord` on init, and then _directly pass the relevant model attributes where needed_... but since there are a lot of munging behaviors that need to happen (parsing dates, extract information from the location field), I would probably wrap all of this into a separate component and create actions for each field. My main concern is when someone might want to go back and edit a hearing for some reason — we wouldn't be able to re-use this.

Reply:
hmm okay good point, I'll look into doing that